pkg/scheduler/actions/integration_tests/allocate/allocate_topology_test.go

```diff
@@ -0,0 +1,1055 @@
+						RequiredGPUsPerTask: 1,
+						Priority:            constants.PriorityTrainNumber,
+						QueueName:           "queue0",
+						Topology: &enginev2alpha2.TopologyConstraint{
+							Topology:              "cluster-topology",
+							RequiredTopologyLevel: "k8s.io/rack",
+						},
+						Tasks: []*tasks_fake.TestTaskBasic{
+							{
+								State: pod_status.Pending,
+							},
+							{
+								State: pod_status.Pending,
+							},
+							{
+								State: pod_status.Pending,
+							},
+							{
+								State: pod_status.Pending,
+							},
+						},
+					},
+				},
+				Nodes: map[string]nodes_fake.TestNodeBasic{
+					"node0": {
+						GPUs: 2,
+						Labels: map[string]string{
+							"k8s.io/rack":  "rack1",
+							"k8s.io/spine": "spine1",
+						},
+					},
+					"node1": {
+						GPUs: 2,
+						Labels: map[string]string{
+							"k8s.io/rack":  "rack1",
+							"k8s.io/spine": "spine2",
+						},
+					},
+					"node2": {
+						GPUs: 2,
+						Labels: map[string]string{
+							"k8s.io/rack":  "rack2",
+							"k8s.io/spine": "spine1",
+						},
+					},
+					"node3": {
+						GPUs: 2,
+						Labels: map[string]string{
+							"k8s.io/rack":  "rack2",
+							"k8s.io/spine": "spine2",
+						},
+					},
+				},
+				Queues: []test_utils.TestQueueBasic{
+					{
+						Name:               "queue0",
+						ParentQueue:        "department-a",
+						DeservedGPUs:       8,
+						GPUOverQuotaWeight: 1,
+						MaxAllowedGPUs:     8,
+					},
+				},
+				Departments: []test_utils.TestDepartmentBasic{
+					{
+						Name:         "department-a",
+						DeservedGPUs: 8,
+					},
+				},
+				TaskExpectedResults: map[string]test_utils.TestExpectedResultBasic{
+					"running_job0-0": {
+						GPUsRequired:         1,
+						Status:               pod_status.Running,
+						DontValidateGPUGroup: true,
+					},
+					"running_job1-0": {
+						GPUsRequired:         1,
+						Status:               pod_status.Running,
+						DontValidateGPUGroup: true,
+					},
+					"pending_job0-0": {
+						GPUsRequired:         1,
+						Status:               pod_status.Pending,
+						DontValidateGPUGroup: true,
+					},
+					"pending_job0-1": {
+						GPUsRequired:         1,
+						Status:               pod_status.Pending,
+						DontValidateGPUGroup: true,
+					},
+					"pending_job0-2": {
+						GPUsRequired:         1,
+						Status:               pod_status.Pending,
+						DontValidateGPUGroup: true,
+					},
+					"pending_job0-3": {
+						GPUsRequired:         1,
+						Status:               pod_status.Pending,
+						DontValidateGPUGroup: true,
+					},
+				},
+				Mocks: &test_utils.TestMock{
+					CacheRequirements: &test_utils.CacheMocking{
+						NumberOfCacheBinds: 4,
+					},
+				},
+			},
+			RoundsUntilMatch: 1,
+		},
+		{
+			TestTopologyBasic: test_utils.TestTopologyBasic{
+				Name: "Prefer topology level - schedule all tasks on capable domain (even though first domain can have part of tasks)",
```

Comment:
Suggesting a shorter name:
"Preferred topology - schedule job on the same rack"